Sources/CvUnit.cpp

```diff
@@ -7026,6 +7026,22 @@ bool CvUnit::canScrap() const
 }
 
 
+int CvUnit::calculateScrapValue() const
+{
+	float fCost = getUnitInfo().getProductionCost() * GC.getGameSpeedInfo(GC.getGame().getGameSpeedType()).getTrainPercent() / 100.0f;
+	fCost += fCost * getUpgradeDiscount() / 100.0f;
+
+	if (GC.getGame().isOption(GAMEOPTION_SIZE_MATTERS))
+	{
+		int iGroupDiff = groupRank() - m_pUnitInfo->getBaseGroupRank();
+		// pow(int, int) returns int, not double...
+		fCost *= std::pow(3.0f, iGroupDiff);
+	}
```

Comment:
👍 / 👎 ?

Reply:
> 👍 / 👎 ?
> I have no internal sense of perf issues lol. Is using pow(double, int) better than pow(float, int) and such, or float x = int / double vs or float x = int / float, etc? Doubt it matters, but don't know 🤷

Try to not ever use any floating point math where game state is concerned as it can cause OOS issues, multiply with 100 before starting and divide by 100 at the end.

Reply:
made a PR to this PR that resolves the OOS potential of this code.

Reply:
Oof, that is right. Will look at my other commits to cpp code, may have made same mistake elsewhere...
If you're done fixing all my terrible code (for which I am eternally grateful...) should I merge the PR(s), or are you still looking at things?

Reply:
Merge away if you have no objections, I'm all done I think.

I don't have write permission, so I can't merge the PR.